### PR TITLE
[Settings UI] Allow renaming a profile

### PR DESCRIPTION
**Pull request by carlos-zamora**
This adds a "Name" text box to the Profile page in the Settings UI.
Any changes to the name/icon are propagated to the relevant
NavigationViewItem.

Base Layer does not have a "Name".

## References
#6800 - Settings UI Epic

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 1]
```diff
@@ -373,6 +373,21 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         icon.IconSource(iconSource);
         profileNavItem.Icon(icon);
 
+        // Update the menu item when the icon/name changes
+        profile.PropertyChanged([profileNavItem](const auto&, const WUX::Data::PropertyChangedEventArgs& args) {
```
[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:377]: You can `winrt::make_weak(profileNavItem)` on it. The only reason it's required is that ProfileNavItem's tag is the ProfileViewModel (+1 reference; it keeps it alive), and ProfileViewModel's event handler (here) keeps ProfileNavItem alive (+1 reference). This means that these two objects can _never_ be deleted. They refer to eachother indefinitely

[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:377]: It's a big ole memory leak.

**Review by hafixo — APPROVED** [review]
#8837

**Review by hafixo — CHANGES_REQUESTED** [review]
#8837 
Merge pull requested

**Review by zadjii-msft — APPROVED** [review]

**Review by DHowett — APPROVED** [review]
Love it

**Comment by DHowett** [comment]
Profile Renaming Demo

ProfileRenamingDemo2

**Pull request merged**